powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/cache/DataStore.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.parameters.cache;
+
+import java.time.Instant;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Internal store used to cache parameters
+ */
+public class DataStore {
+
+    private final ConcurrentHashMap<String, ValueNode> store;
+    private final NowProvider nowProvider;
+
+    public DataStore(NowProvider nowProvider) {
```

Comment:
Instead of having NowProvider can these classes depend on Clock instead ? That is the java8 recommended way of dealing with manipulation of time

Reply:
OK, I did that. Looks better now.